Cannot build on Windows

It fails to build on Windows:

```bash
error[E0433]: failed to resolve: could not find `sys` in `nix`
  --> D:\packages\cargo\registry\src\index.crates.io-1949cf8c6b5b557f\wrkflw-utils-0.7.0\src\lib.rs:40:14
   |
40 |     use nix::sys::stat::Mode;
   |              ^^^ could not find `sys` in `nix`

error[E0432]: unresolved import `nix::fcntl`
  --> D:\packages\cargo\registry\src\index.crates.io-1949cf8c6b5b557f\wrkflw-utils-0.7.0\src\lib.rs:39:14
   |
39 |     use nix::fcntl::{open, OFlag};
   |              ^^^^^ could not find `fcntl` in `nix`

error[E0433]: failed to resolve: could not find `unix` in `os`
  --> D:\packages\cargo\registry\src\index.crates.io-1949cf8c6b5b557f\wrkflw-utils-0.7.0\src\lib.rs:43:18
   |
43 |     use std::os::unix::io::RawFd;
   |                  ^^^^ could not find `unix` in `os`
   |
note: found an item that was configured out
  --> D:\packages\rustup\toolchains\stable-x86_64-pc-windows-msvc\lib/rustlib/src/rust\library\std\src\os\mod.rs:36:9
   |
36 | pub mod unix {}
   |         ^^^^
note: the item is gated here
  --> D:\packages\rustup\toolchains\stable-x86_64-pc-windows-msvc\lib/rustlib/src/rust\library\std\src\os\mod.rs:28:1
   |
28 | / #[cfg(all(
29 | |     doc,
30 | |     any(
31 | |         all(target_arch = "wasm32", not(target_os = "wasi")),
...  |
34 | | ))]
   | |___^
note: found an item that was configured out
  --> D:\packages\rustup\toolchains\stable-x86_64-pc-windows-msvc\lib/rustlib/src/rust\library\std\src\os\mod.rs:85:9
   |
85 | pub mod unix;
   |         ^^^^
note: the item is gated here
  --> D:\packages\rustup\toolchains\stable-x86_64-pc-windows-msvc\lib/rustlib/src/rust\library\std\src\os\mod.rs:84:1
   |
84 | #[cfg(all(not(target_os = "hermit"), any(unix, doc)))]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0432]: unresolved import `nix::unistd`
  --> D:\packages\cargo\registry\src\index.crates.io-1949cf8c6b5b557f\wrkflw-utils-0.7.0\src\lib.rs:41:14
   |
41 |     use nix::unistd::{close, dup, dup2};
   |              ^^^^^^ could not find `unistd` in `nix`

Some errors have detailed explanations: E0432, E0433.
For more information about an error, try `rustc --explain E0432`.
error: could not compile `wrkflw-utils` (lib) due to 4 previous errors
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -3,7 +3,7 @@ name: Build
 on:
   workflow_dispatch:
   push:
-    branches: [ main ]
+    branches: [main]
   pull_request:
 
 jobs:
@@ -12,12 +12,14 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ubuntu-latest, macos-latest]
+        os: [ubuntu-latest, macos-latest, windows-latest]
         include:
           - os: ubuntu-latest
             target: x86_64-unknown-linux-gnu
           - os: macos-latest
             target: x86_64-apple-darwin
+          - os: windows-latest
+            target: x86_64-pc-windows-msvc
 
     steps:
       - name: Checkout code
@@ -31,27 +33,27 @@ jobs:
           target: ${{ matrix.target }}
           override: true
           components: clippy, rustfmt
-      
+
       - name: Check formatting
         uses: actions-rs/cargo@v1
         with:
           command: fmt
           args: -- --check
-      
+
       - name: Run clippy
         uses: actions-rs/cargo@v1
         with:
           command: clippy
           args: -- -D warnings
-      
+
       - name: Build
         uses: actions-rs/cargo@v1
         with:
           command: build
           args: --target ${{ matrix.target }}
-      
+
       - name: Run tests
         uses: actions-rs/cargo@v1
         with:
           command: test
-          args: --target ${{ matrix.target }} 
+          args: --target ${{ matrix.target }}

--- a/crates/runtime/src/emulation.rs
+++ b/crates/runtime/src/emulation.rs
@@ -793,7 +793,7 @@ async fn cleanup_processes() {
             let _ = Command::new("taskkill")
                 .arg("/F")
                 .arg("/PID")
-                .arg(&pid.to_string())
+                .arg(pid.to_string())
                 .output();
         }
 

--- a/crates/utils/Cargo.toml
+++ b/crates/utils/Cargo.toml
@@ -17,4 +17,6 @@ wrkflw-models = { path = "../models", version = "0.7.0" }
 # External dependencies
 serde.workspace = true
 serde_yaml.workspace = true
+
+[target.'cfg(unix)'.dependencies]
 nix.workspace = true

--- a/crates/utils/README.md
+++ b/crates/utils/README.md
@@ -3,7 +3,7 @@
 Shared helpers used across crates.
 
 - Workflow file detection (`.github/workflows/*.yml`, `.gitlab-ci.yml`)
-- File-descriptor redirection utilities for silencing noisy subprocess output
+- File-descriptor redirection utilities for silencing noisy subprocess output (Unix only; Windows support is limited)
 
 ### Example
 
@@ -14,7 +14,7 @@ use wrkflw_utils::{is_workflow_file, fd::with_stderr_to_null};
 assert!(is_workflow_file(Path::new(".github/workflows/ci.yml")));
 
 let value = with_stderr_to_null(|| {
-    eprintln!("this is hidden");
+    eprintln!("this is hidden on Unix, visible on Windows");
     42
 }).unwrap();
 assert_eq!(value, 42);

--- a/crates/utils/src/lib.rs
+++ b/crates/utils/src/lib.rs
@@ -43,7 +43,7 @@ pub fn is_workflow_file(path: &Path) -> bool {
 /// differences in file descriptor handling. The functions will execute without
 /// error but stderr may not be fully suppressed.
 pub mod fd {
-    use std::io::{self, Result};
+    use std::io::Result;
 
     /// Represents a redirected stderr that can be restored
     pub struct RedirectedStderr {
@@ -61,6 +61,7 @@ pub mod fd {
         use nix::fcntl::{open, OFlag};
         use nix::sys::stat::Mode;
         use nix::unistd::{close, dup, dup2};
+        use std::io;
         use std::os::unix::io::RawFd;
         use std::path::Path;
 
@@ -119,8 +120,6 @@ pub mod fd {
     #[cfg(windows)]
     mod windows_impl {
         use super::*;
-        use std::fs::OpenOptions;
-        use std::io::Write;
 
         impl RedirectedStderr {
             /// Creates a new RedirectedStderr that redirects stderr to NUL on Windows

--- a/crates/utils/src/lib.rs
+++ b/crates/utils/src/lib.rs
@@ -35,78 +35,146 @@ pub fn is_workflow_file(path: &Path) -> bool {
 }
 
 /// Module for safely handling file descriptor redirection
+///
+/// On Unix systems (Linux, macOS), this module provides true file descriptor
+/// redirection by duplicating stderr and redirecting it to /dev/null.
+///
+/// On Windows systems, the redirection functionality is limited due to platform
+/// differences in file descriptor handling. The functions will execute without
+/// error but stderr may not be fully suppressed.
 pub mod fd {
-    use nix::fcntl::{open, OFlag};
-    use nix::sys::stat::Mode;
-    use nix::unistd::{close, dup, dup2};
     use std::io::{self, Result};
-    use std::os::unix::io::RawFd;
-    use std::path::Path;
-
-    /// Standard file descriptors
-    const STDERR_FILENO: RawFd = 2;
 
     /// Represents a redirected stderr that can be restored
     pub struct RedirectedStderr {
-        original_fd: Option<RawFd>,
-        null_fd: Option<RawFd>,
+        #[cfg(unix)]
+        original_fd: Option<std::os::unix::io::RawFd>,
+        #[cfg(unix)]
+        null_fd: Option<std::os::unix::io::RawFd>,
+        #[cfg(windows)]
+        _phantom: std::marker::PhantomData<()>,
     }
 
-    impl RedirectedStderr {
-        /// Creates a new RedirectedStderr that redirects stderr to /dev/null
-        pub fn to_null() -> Result<Self> {
-            // Duplicate the current stderr fd
-            let stderr_backup = match dup(STDERR_FILENO) {
-                Ok(fd) => fd,
-                Err(e) => return Err(io::Error::other(e)),
-            };
+    #[cfg(unix)]
+    mod unix_impl {
+        use super::*;
+        use nix::fcntl::{open, OFlag};
+        use nix::sys::stat::Mode;
+        use nix::unistd::{close, dup, dup2};
+        use std::os::unix::io::RawFd;
+        use std::path::Path;
 
-            // Open /dev/null
-            let null_fd = match open(Path::new("/dev/null"), OFlag::O_WRONLY, Mode::empty()) {
-                Ok(fd) => fd,
-                Err(e) => {
+        /// Standard file descriptors
+        const STDERR_FILENO: RawFd = 2;
+
+        impl RedirectedStderr {
+            /// Creates a new RedirectedStderr that redirects stderr to /dev/null
+            pub fn to_null() -> Result<Self> {
+                // Duplicate the current stderr fd
+                let stderr_backup = match dup(STDERR_FILENO) {
+                    Ok(fd) => fd,
+                    Err(e) => return Err(io::Error::other(e)),
+                };
+
+                // Open /dev/null
+                let null_fd = match open(Path::new("/dev/null"), OFlag::O_WRONLY, Mode::empty()) {
+                    Ok(fd) => fd,
+                    Err(e) => {
+                        let _ = close(stderr_backup); // Clean up on error
+                        return Err(io::Error::other(e));
+                    }
+                };
+
+                // Redirect stderr to /dev/null
+                if let Err(e) = dup2(null_fd, STDERR_FILENO) {
                     let _ = close(stderr_backup); // Clean up on error
+                    let _ = close(null_fd);
                     return Err(io::Error::other(e));
                 }
-            };
 
-            // Redirect stderr to /dev/null
-            if let Err(e) = dup2(null_fd, STDERR_FILENO) {
-                let _ = close(stderr_backup); // Clean up on error
-                let _ = close(null_fd);
-                return Err(io::Error::other(e));
+                Ok(RedirectedStderr {
+                    original_fd: Some(stderr_backup),
+                    null_fd: Some(null_fd),
+                })
             }
-
-            Ok(RedirectedStderr {
-                original_fd: Some(stderr_backup),
-                null_fd: Some(null_fd),
-            })
         }
-    }
 
-    impl Drop for RedirectedStderr {
-        /// Automatically restores stderr when the RedirectedStderr is dropped
-        fn drop(&mut self) {
-            if let Some(orig_fd) = self.original_fd.take() {
-                // Restore the original stderr
-                let _ = dup2(orig_fd, STDERR_FILENO);
-                let _ = close(orig_fd);
-            }
+        impl Drop for RedirectedStderr {
+            /// Automatically restores stderr when the RedirectedStderr is dropped
+            fn drop(&mut self) {
+                if let Some(orig_fd) = self.original_fd.take() {
+                    // Restore the original stderr
+                    let _ = dup2(orig_fd, STDERR_FILENO);
+                    let _ = close(orig_fd);
+                }
 
-            // Close the null fd
-            if let Some(null_fd) = self.null_fd.take() {
-                let _ = close(null_fd);
+                // Close the null fd
+                if let Some(null_fd) = self.null_fd.take() {
+                    let _ = close(null_fd);
+                }
             }
         }
     }
 
-    /// Run a function with stderr redirected to /dev/null, then restore stderr
+    #[cfg(windows)]
+    mod windows_impl {
+        use super::*;
+        use std::fs::OpenOptions;
+        use std::io::Write;
+
+        impl RedirectedStderr {
+            /// Creates a new RedirectedStderr that redirects stderr to NUL on Windows
+            pub fn to_null() -> Result<Self> {
+                // On Windows, we can't easily redirect stderr at the file descriptor level
+                // like we can on Unix systems. This is a simplified implementation that
+                // doesn't actually redirect but provides the same interface.
+                // The actual stderr suppression will need to be handled differently on Windows.
+                Ok(RedirectedStderr {
+                    _phantom: std::marker::PhantomData,
+                })
+            }
+        }
+
+        impl Drop for RedirectedStderr {
+            /// No-op drop implementation for Windows
+            fn drop(&mut self) {
+                // Nothing to restore on Windows in this simplified implementation
+            }
+        }
+    }
+
+    /// Run a function with stderr redirected to /dev/null (Unix) or suppressed (Windows), then restore stderr
+    ///
+    /// # Platform Support
+    /// - **Unix (Linux, macOS)**: Fully supported - stderr is redirected to /dev/null
+    /// - **Windows**: Limited support - function executes but stderr may be visible
+    ///
+    /// # Example
+    /// ```
+    /// use wrkflw_utils::fd::with_stderr_to_null;
+    ///
+    /// let result = with_stderr_to_null(|| {
+    ///     eprintln!("This will be hidden on Unix");
+    ///     42
+    /// }).unwrap();
+    /// assert_eq!(result, 42);
+    /// ```
     pub fn with_stderr_to_null<F, T>(f: F) -> Result<T>
     where
         F: FnOnce() -> T,
     {
-        let _redirected = RedirectedStderr::to_null()?;
-        Ok(f())
+        #[cfg(unix)]
+        {
+            let _redirected = RedirectedStderr::to_null()?;
+            Ok(f())
+        }
+        #[cfg(windows)]
+        {
+            // On Windows, we can't easily redirect stderr at the FD level,
+            // so we just run the function without redirection.
+            // This means stderr won't be suppressed on Windows, but the function will work.
+            Ok(f())
+        }
     }
 }
 
@@ -116,15 +184,16 @@ mod tests {
 
     #[test]
     fn test_fd_redirection() {
-        // This test will write to stderr, which should be redirected
+        // This test will write to stderr, which should be redirected on Unix
+        // On Windows, it will just run normally without redirection
         let result = fd::with_stderr_to_null(|| {
-            // This would normally appear in stderr
-            eprintln!("This should be redirected to /dev/null");
+            // This would normally appear in stderr (suppressed on Unix, visible on Windows)
+            eprintln!("This should be redirected to /dev/null on Unix");
             // Return a test value to verify the function passes through the result
             42
         });
 
-        // The function should succeed and return our test value
+        // The function should succeed and return our test value on both platforms
         assert!(result.is_ok());
         assert_eq!(result.unwrap(), 42);
     }

